File: codeforces/499B/Lecture.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

int main() {
  int n, m;
  cin >> n >> m;

  map<string, string> language;

  for(int i = 0; i < m; i++) {
    string first, second;
    cin >> first >> second;

    string prefer = (first.length() <= second.length()) ? first : second;

    language[first] = prefer;
  }

  for(int i = 0; i < n; i++) {
    string word;
    cin >> word;

    auto chosenWord = language.find(word);

    cout << chosenWord->second;
  }
  cout << "\n";
}

File: codeforces/499B/Lecture.java
import java.util.Scanner;
import java.util.Map;
import java.util.TreeMap;

public class Lecture {
  public static void main(String [] args) {
    Scanner input = new Scanner(System.in);
    int n = input.nextInt();
    int m = input.nextInt();

    Map<String, String> languages = new TreeMap<String, String>();

    for(int i = 0; i < m; i++) {
      String first = input.next();
      String second = input.next();

      String prefer = (first.length() <= second.length()) ? first : second;

      languages.put(first, prefer);
    }

    for(int i = 0; i < n; i++) {
      System.out.print(languages.get(input.next()) + " ");
    }
    System.out.println();
  }
}
      
